src/assets/CollectiblesController.ts

```diff
@@ -1148,6 +1153,105 @@ export class CollectiblesController extends BaseController<
 
     this.updateNestedCollectibleState(collectibles, ALL_COLLECTIBLES_STATE_KEY);
   }
+
+  /**
+   * Update colletible isTransacted status
+   *
+   * @param address - Hex address of the collectible contract.
+   * @param tokenId - Number that represents the id of the token.
+   * @param isTransacting - Collectible transacting status.
+   * @param transactionId - Hex id of the send transaction of the collectible.
+   */
+  updateCollectibleTransactionStatus(
+    address: string,
+    tokenId: string,
+    isTransacting: boolean,
+    transactionId: string,
+  ) {
+    const { allCollectibles } = this.state;
+    const { chainId, selectedAddress } = this.config;
+    const collectibles = allCollectibles[selectedAddress]?.[chainId] || [];
+    const index: number = collectibles.findIndex(
+      (collectible) =>
+        collectible.address === address && collectible.tokenId === tokenId,
+    );
+
+    if (index === -1) {
+      return;
+    }
+    const updatedCollectible: Collectible = {
+      ...collectibles[index],
+      isTransacting,
+      transactionId,
+    };
+
+    // Update Collectibles array
+    collectibles[index] = updatedCollectible;
+
+    this.updateNestedCollectibleState(collectibles, ALL_COLLECTIBLES_STATE_KEY);
+  }
+  /**
+   * Reset the collectible transaction id value and turn false the isTransacting prop
+   *
+   * @param transactionId - Hex id of the send transaction of the collectible.
+   * @returns
+   */
+  resetCollectibleTransactionStatusByTransactionId(
```

Comment:
What if this method were more generic too? Maybe something like `updateCollectible` instead. This could work one of two ways:

1. You pass an address, token id, and a collectible, and it calls ``findCollectibleByAddressAndTokenId(address, tokenId)` to find the *current* version of the collectibles in the list and update it with the *new* collectible.
2. You pass a collectible object that you've already modified, and it would look up the collectible using `findCollectibleByAddressAndTokenId` and then effectively replace that collectible in the list.

Reply:
The obstacle that I found in this method is that I really need to find the collectible by transactionId in the transactions list view, but I will discuss this in a team sprint review to see if it is possible to find it with the address and token Id!
Thank you so much for the review!